src/librustc_mir/build/matches/mod.rs

```diff
@@ -266,6 +266,7 @@ enum TestKind<'tcx> {
     // test the branches of enum
     Switch {
         adt_def: AdtDef<'tcx>,
+        variants: Vec<bool>,
```

Comment:
maybe a good idea to add a comment explaining what this vector represents, when it is true, etc